# MEGA UE5: Machine Learning
In this exercise, you will use Python to explore and get insights into a large cohort of COVID patient
cases using traditional machine learning approaches for stratification (i.e., dimensionality
reduction and clustering). You will work with EHR data (`data_ehr.csv`) that you can find in the UE5-
data.zip file. There is also a data_description.csv that includes a description of the variables
included in this EHR data set.

For the submission, you need to provide two deliverables:
1. A `.pdf` with the answers to the questions (a-f) below.
2. Your `.py` script(s).

## Part 1: Data Preprocessing
Read the EHR data (data_ehr.csv), for instance with the Pandas library. The dataset contains
missing values that are encoded with NA or empty values. Clean the dataset by removing columns
where more than 30% of the values are NA.

a- How many dimensions (attributes) and instances (cases or number of patients) does
the clean dataset contain? How many dimensions and instances did it contain before the cleaning?

b- What type of data does the clean dataset contain?

In [13]:
import part1 as preprocessing

data = preprocessing.read_csv('UE5-data/data_ehr.csv')
rows, columns = data.shape
print('Rows:', rows, 'Columns:', columns)

clean_data = preprocessing.clean_data(data)
clean_rows, clean_columns = clean_data.shape
print('Rows_cleaned:', clean_rows, 'Columns_cleaned:', clean_columns)

Rows: 1384 Columns: 130
Rows_cleaned: 1384 Columns_cleaned: 118


The file data_ehr.csv contains 130 dimensions (columns) and 1384 instances (rows). After preprocessing and removing columns/dimensions where more than 30% of the values are NA it only contains 118 dimensions (columns). The number of instances stays the same since removal only concerned columns.

In [4]:
print(clean_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1384 entries, 0 to 1383
Columns: 118 entries, to_patient_id to therapeutic.heparin.Boolean
dtypes: bool(3), float64(32), int64(1), object(82)
memory usage: 1.2+ MB
None


In the preprocessed "cleaned" data there are 118 columns left. Each of these has a specific data type: 3 columns-bool, 32 columns-float64, 1 columns-int64, 82 columns-object